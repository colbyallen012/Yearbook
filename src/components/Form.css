p {
  background-color: white;
  border-radius: 20px;
  color: blue;
  font-size: 20px;
  font-weight: bold;
  margin: auto;
  padding: 2px;
  width: 50%;
}

input {
  border-radius: 20px;
  margin: 10px;
}