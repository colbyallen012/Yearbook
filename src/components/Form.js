import React, {Component} from 'react';
import './Form.css'

class Form extends Component {
  constructor() {
    super();
    this.state = {
      name: '',
      quote: '',
      superlative: ''
    }
  }

  handleChange = event => {
    this.setState({[event.target.name]: event.target.value})
  }

  submitStudent = event => {
    event.preventDefault();
    const newStudents = {
      id: Date.now(),
      ...this.state
    }
    this.props.addStudents(newStudents);
    this.clearInputs()
  }

  clearInputs = () => {
    this.setState({ name: '', quote: '', superlative: '' });
  }

  render() {
    return (
      <form>
        <input 
          type='text'
          placeholder='Name'
          name='name'
          value={this.state.name}
          onChange={this.handleChange}
        />

        <input 
          type='text'
          placeholder='Quote'
          name='quote'
          value={this.state.quote}
          onChange={this.handleChange}
        />

        <input 
          type='text'
          placeholder='Superlative'
          name='superlative'
          value={this.state.superlative}
          onChange={this.handleChange}
        />

        <button onClick={event => this.submitStudent(event)}>Submit</button>

      </form>
    )
  }
}

export default Form;

