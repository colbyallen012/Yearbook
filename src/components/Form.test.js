import React from 'react';
import { shallow, mount } from 'enzyme';
import Form from './Form.js';


describe('Form', () => {
  it('should match the snapshot with all the data passed in correctly', () => {
    const wrapper = shallow(
      <Form
        name='Colby'
        quote='#grindin'
        superlative='most likely to eat all the za'
      />
    );

    expect(wrapper).toMatchSnapshot()
  });
});