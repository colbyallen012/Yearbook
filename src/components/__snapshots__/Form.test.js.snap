// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should match the snapshot with all the data passed in correctly 1`] = `
<form>
  <input
    name="name"
    onChange={[Function]}
    placeholder="Name"
    type="text"
    value=""
  />
  <input
    name="quote"
    onChange={[Function]}
    placeholder="Quote"
    type="text"
    value=""
  />
  <input
    name="superlative"
    onChange={[Function]}
    placeholder="Superlative"
    type="text"
    value=""
  />
  <button
    onClick={[Function]}
  >
    Submit
  </button>
</form>
`;
